pandas/core/frame.py

```diff
@@ -3620,6 +3620,11 @@ def reindexer(value):
             value = cast_scalar_to_array(len(self.index), value)
             value = maybe_cast_to_datetime(value, infer_dtype)
 
+        # convert pandas array to numpy array
+        if isinstance(value, ABCPandasArray):
+            value = value.to_numpy()
+            return np.atleast_2d(np.asarray(value))
```

Comment:
Why the `np.asarray`?

Reply:
And do you even need this early return? Can you not just do `value = value.to_numpy()`?

Reply:
The reason for `np.asarray` was because the same was done if a Numpy array was passed to the column.

You're right though, I shouldn't be returning this early, I had a misconception that the next check would return the wrong value

```
# return internal types directly
        if is_extension_type(value) or is_extension_array_dtype(value):
            return value
```

Will fix this.

Reply:
I guess putting the conversion in `make_blocks` is a better idea, it was the first thing that came to mind, but I thought it to be a sanitary process, just like the conversion of NumpyArray to a 2D Format.

Anyway I'll make the requisite changes